test/test_images/sender/main.go

```diff
@@ -129,15 +121,49 @@ func writeTerminationMessage(result interface{}) error {
 
 // isRetryable determines if the err is an error which is retryable
 func isRetryable(err error) bool {
-	var httpResult *cehttp.Result
-	if errors.As(err, &httpResult) {
+	var result *cehttp.Result
+	if protocol.ResultAs(err, &result) {
 		// Potentially retry when:
 		// - 404 Not Found
 		// - 500 Internal Server Error, it is currently for reducing flakiness caused by Workload Identity credential sync up.
 		// We should remove it after https://github.com/google/knative-gcp/issues/1058 lands, as 500 error may indicate bugs in our code.
 		// - 503 Service Unavailable (with or without Retry-After) (IGNORE Retry-After)
-		sc := httpResult.StatusCode
-		return sc == 404 || sc == 500 || sc == 503
+		sc := result.StatusCode
+		if sc == 404 || sc == 500 || sc == 503 {
+			log.Printf("got error: %s, retry sending event. \n", result.Error())
+			return true
+		}
 	}
 	return false
 }
+
+func newDefaultClient(ctx context.Context, target ...string) (context.Context, cev2.Client, error) {
```

Comment:
Would like to eventually use knclient in sender image, however, knclient is using cev1 and changing knclient to cev2 needs changes in other images (after #1335 merged, we are using knclient for most of the images). Created a follow-up issue for that: https://github.com/google/knative-gcp/issues/1467